### PR TITLE
Cleanup some outdated helpers in libraries/Common

### PR DESCRIPTION
**Pull request by huoyaoyuan**
Replace with public API available in newer version.

### CODE REVIEW

File: src/libraries/Common/src/System/Net/CookieParser.cs [hunk 1]
```diff
@@ -140,7 +140,7 @@ internal string Extract()
             {
                 tokenString = Quoted ?
                     _tokenStream.Substring(_start, _tokenLength) :
-                    _tokenStream.SubstringTrim(_start, _tokenLength);
+                    _tokenStream.AsSpan().Slice(_start, _tokenLength).Trim().ToString();
```
Comment (huoyaoyuan) [inline src/libraries/Common/src/System/Net/CookieParser.cs:143]:
```suggestion
                    _tokenStream.AsSpan(_start, _tokenLength).Trim().ToString();
```

File: src/libraries/System.Net.Mail/src/System/Net/Mail/MailAddressParser.cs [hunk 2]
```diff
@@ -421,7 +421,7 @@ private static bool TryParseDisplayName(string data, ref int index, bool expectM
 
                 // Do not include the Comma (if any), and because there were no bounding quotes,
                 // trim extra whitespace.
-                displayName = data.SubstringTrim(index + 1, startingIndex - index);
+                displayName = data.AsSpan().Slice(index + 1, startingIndex - index).Trim().ToString();
```
Comment (huoyaoyuan) [inline src/libraries/System.Net.Mail/src/System/Net/Mail/MailAddressParser.cs:424]:
```suggestion
                displayName = data.AsSpan(index + 1, startingIndex - index).Trim().ToString();
```

**Comment by huoyaoyuan** [comment]
CI failure is weird. Seems unrelated.

**Comment by marek-safar** [comment]
@stephentoub could you have a look?

**Review by adamsitnik — APPROVED** [review]
Overall it LGTM, thank you for your contribution @huoyaoyuan !

The failing test seems to be unrelated, but it has "0.00% failure rate" so I am going to re-run it to be sure.

**Pull request merged**